### PR TITLE
Introduce extensible payloads

### LINKED ISSUE #1663: Add extensible payloads and use them for consensus
Very controversial move, but see neo-project/neo#2101, neo-project/neo#2209 and neo-project/neo#2202.

### PR DESCRIPTION
**Pull request by fyrchik**
Close #1663 .

Sender whitelist will be done in another PR.

### CODE REVIEW

File: pkg/network/server.go [hunk 1]
```diff
@@ -715,10 +715,29 @@ func (s *Server) handleGetHeadersCmd(p Peer, gh *payload.GetBlockByIndex) error
...
 
-// handleConsensusCmd processes received consensus payload.
-// It never returns an error.
-func (s *Server) handleConsensusCmd(cp *consensus.Payload) error {
-	s.consensus.OnPayload(cp)
+const extensibleVerifyMaxGAS = 2000000
+
+// handleExtensibleCmd processes received extensible payload.
+func (s *Server) handleExtensibleCmd(e *payload.Extensible) error {
+	if err := s.chain.VerifyWitness(e.Sender, e, &e.Witness, extensibleVerifyMaxGAS); err != nil {
```
Comment (roman-khimov) [inline pkg/network/server.go:722]: Isn't it duplicating signature check in `validatePayload`?

File: pkg/consensus/consensus.go [hunk 2]
```diff
@@ -52,11 +52,11 @@ type Service interface {
 	Shutdown()
 
 	// OnPayload is a callback to notify Service about new received payload.
-	OnPayload(p *Payload)
+	OnPayload(p *npayload.Extensible)
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/consensus/consensus.go:55]: We probably can but it is not straightforward.
1. Validator index should be verified based on `Sender`.
2. DBFT itself uses payload hashes which are in turn calculated based on whole extensible payload.

I'd rather leave it as it is now, possibly updating in future.

File: pkg/network/server.go (hunk 1 above)
Comment (fyrchik) [reply pkg/network/server.go:722]: Yes, fixed `validatePayload`.

**Comment by roman-khimov** [comment]
Please merge dbft (and update its revision here) and this will good to merge.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**